sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusScaleMonitor.cs

```diff
@@ -42,20 +49,52 @@ public ServiceBusScaleMonitor(string functionId, ServiceBusEntityType serviceBus
             _nextWarningTime = DateTime.UtcNow;
         }
 
-        public ScaleMonitorDescriptor Descriptor
+        /// <summary>
+        /// Public constructor used to retrieve metrics and make scale decisions for Service Bus entities.
+        /// </summary>
+        /// <param name="functionId">The function name to make scale decisions for.</param>
+        /// <param name="serviceBusEntityType">Either a Queue or Topic. <see cref="ServiceBusEntityType"/></param>
+        /// <param name="entityPath">The name of the queue or topic that scale decisions are made for.</param>
+        /// <param name="connection">The connection name used for the function.</param>
+        /// <param name="loggerFactory">Used to create an ILogger instance.</param>
+        /// <param name="configuration">IConfiguration object that contains connection information specified by the "connection" name.</param>
+        /// <param name="componentFactory">Contains logic used for identity-based connections.</param>
+        /// <param name="messagingProvider">Used for creation of Service Bus message processing entities.</param>
+        /// <param name="logForwarder">Used to forward log events from Azure SDK EventSources to a logger factory.</param>
+        /// <param name="options">Configuration options used by the Service Bus extension.</param>
+        public ServiceBusScaleMonitor(string functionId, ServiceBusEntityType serviceBusEntityType, string entityPath, string connection, ILoggerFactory loggerFactory, IConfiguration configuration, AzureComponentFactory componentFactory, MessagingProvider messagingProvider, AzureEventSourceLogForwarder logForwarder, IOptions<ServiceBusOptions> options)
+        {
+            _functionId = functionId;
+            _serviceBusEntityType = serviceBusEntityType;
+            _entityPath = entityPath;
+            _scaleMonitorDescriptor = new ScaleMonitorDescriptor($"{_functionId}-ServiceBusTrigger-{_entityPath}".ToLower(CultureInfo.InvariantCulture));
+            _isListeningOnDeadLetterQueue = entityPath.EndsWith(DeadLetterQueuePath, StringComparison.OrdinalIgnoreCase);
+            _logger = loggerFactory.CreateLogger<ServiceBusScaleMonitor>();
+            _nextWarningTime = DateTime.UtcNow;
+            _clientFactory = new Lazy<ServiceBusClientFactory>(() => new ServiceBusClientFactory(configuration, componentFactory, messagingProvider, logForwarder, options));
+            _administrationClient = new Lazy<ServiceBusAdministrationClient>(() => _clientFactory.Value.CreateAdministrationClient(connection));
+            _client = new Lazy<ServiceBusClient>(() => _clientFactory.Value.CreateClientFromSetting(connection));
+            _receiver = new Lazy<ServiceBusReceiver>(() => _client.Value.CreateReceiver(entityPath));
+        }
+
+        ScaleMonitorDescriptor IScaleMonitor.Descriptor
```

Comment:
Why the change here from public to explicit interface implementation? Can't it remain public?